compiler/modules/front.syntax/tests/canonicalTree.bal

```diff
@@ -0,0 +1,295 @@
+// lhs is expected to be the normalized tree node and rhs to be an un-normalized tree node
+function syntaxNodeEquals(SyntaxNode|RootSyntaxNode lhs, SyntaxNode|RootSyntaxNode rhs) returns boolean {
```

Comment:
Why do you need this?

Can't you just normalize and then use `==`?

Reply:
Our SyntaxNode's can have ASTNodes that have positions which we don't change as a part of normalization. So it's possible to have `lhs` and `rhs` to be same (in terms of tokens and their place in the tree) but different in terms of position

Reply:
So the positions in the two trees you are comparing are different?

Where do these two trees come from and why are you comparing them? I suspect there should be a way to set things up so that the trees you are comparing have the same positions.

Reply:
First tree correspond to the ast of the un-normalized version of the code in [`testParse.json`](https://github.com/ballerina-platform/nballerina/blob/main/compiler/modules/front.syntax/tests/data/testParser.json) and the second one corresponds to the normalized version. Objective is the verify in both cases we get the same tree. Previously what we did was run astToString against the ast of the un-normalized version and see if that is same as the normalized version string wise. We can't do that until step 2 is done.